sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/AppendBlobClient.java

```diff
@@ -156,7 +156,8 @@ public AppendBlobItem create(boolean overwrite) {
      *
      * {@codesnippet com.azure.storage.blob.specialized.AppendBlobClient.appendBlock#InputStream-long}
      *
-     * @param data The data to write to the blob.
+     * @param data The data to write to the blob. The data must be markable. This is in order to support retries. If
```

Comment:
Could we enforce the mark supported by calling? 
https://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html#markSupported()